pkg/cloudprovider/vm.go

```diff
@@ -99,6 +99,10 @@ func (c *VMClient) Get(rgName string, nodeName string) (compute.VirtualMachine,
 
 // UpdateIdentities updates the user assigned identities for the provided node
 func (c *VMClient) UpdateIdentities(rg, nodeName string, vm compute.VirtualMachine) error {
+	if *vm.ProvisioningState != string(compute.ProvisioningStateSucceeded) {
```

Comment:
Let me know if there are other states that we want to check

Reply:
I think just waiting for `ProvisioningStateSucceeded` might cause delay in the identity assignment operations. We should check if the `ProvisioningState` is `Deleting` and skip operations only for those.